build/build.go

```diff
@@ -937,13 +937,23 @@ func Build(ctx context.Context, drivers []DriverInfo, opt map[string]Options, do
 											return errors.Errorf("tag is needed when pushing to registry")
 										}
 										pw := progress.ResetTime(pw)
-										for _, name := range strings.Split(pushNames, ",") {
+										pushList := strings.Split(pushNames, ",")
+										for _, name := range pushList {
 											if err := progress.Wrap(fmt.Sprintf("pushing %s with docker", name), pw.Write, func(l progress.SubLogger) error {
 												return pushWithMoby(ctx, d, name, l)
 											}); err != nil {
 												return err
 											}
 										}
+										remoteDigest, err := remoteDigestWithMoby(ctx, d, pushList[0])
+										if err == nil && remoteDigest != "" {
+											if v, ok := rr.ExporterResponse[exptypes.ExporterImageDigestKey]; ok {
+												rr.ExporterResponse[exptypes.ExporterImageConfigDigestKey] = v
```

Comment:
why this? Is this backward compatibility or smth? Why wouldn't `ExporterImageConfigDigestKey` be set already?

Reply:
> Why wouldn't `ExporterImageConfigDigestKey` be set already?

As you can see in https://github.com/crazy-max/buildx-buildkit-tests/runs/5458103650?check_suite_focus=true#step:7:152 (using latest stable buildx version), metadata only returns then following content when image is pushed with moby:

```json
{
  "containerimage.digest": "sha256:1fd302965e74b804eccf7e6ab6ec1edcbd3dfa8b8371ebc7244492832c39dd81"
}
```

I think this is linked to the current implementation on moby: https://github.com/moby/moby/blob/327699c313f51f8062cf3affcb3746f56f3689c6/builder/builder-next/exporter/export.go#L176

Reply:
Ok, we should try to fix this in moby. To make sure we only do this for the old daemons lets only set `ExporterImageConfigDigestKey` when no such value exists and add a comment that this is for handling older daemons that don't return `ExporterImageConfigDigestKey`.